pkg/controller/mysql/user/reconciler.go

```diff
@@ -222,14 +222,13 @@ func (c *external) Observe(ctx context.Context, mg resource.Managed) (managed.Ex
 		return managed.ExternalObservation{}, err
 	}
 
-	cr.SetConditions(xpv1.Available())
-
 	cr.Status.AtProvider.ResourceOptionsAsClauses = resourceOptionsToClauses(observed.ResourceOptions)
 
+	cr.SetConditions(xpv1.Available())
+
 	return managed.ExternalObservation{
-		ResourceExists:          true,
-		ResourceLateInitialized: lateInit(observed, &cr.Spec.ForProvider),
```

Comment:
I mimicked the lateInit from the postgres role reconciler, but apparently I don't understand the implications of this. It caused the User object to never become ready in the integration test. I've removed this in the PR here for now, not sure if needed.